zkevm-circuits/src/rlp_circuit.rs

```diff
@@ -0,0 +1,1705 @@
+                    1.expr(),
+                );
+                cb.require_equal(
+                    "next-to-next tag is Zero => value::Rotation(2) == 128",
+                    meta.query_advice(byte_value, Rotation(2)),
+                    128.expr(),
+                );
+                cb.require_equal(
+                    "next-to-next tag is Zero => value_acc::Rotation(2) == 128",
+                    meta.query_advice(rlp_table.value_acc, Rotation(2)),
+                    0.expr(),
+                );
+
+                // checks for RlpTxTag::RlpLength on the next-to-next-to-next row.
+                cb.require_equal(
+                    "tag::Rotation(3) == RlpTxTag::RlpLength",
+                    meta.query_advice(rlp_table.tag, Rotation(3)),
+                    RlpTxTag::RlpLength.expr(),
+                );
+                cb.require_equal(
+                    "tag_index::Rotation(3) == tag_length::Rotation(3)",
+                    meta.query_advice(rlp_table.tag_rindex, Rotation(3)),
+                    meta.query_advice(tag_length, Rotation(3)),
+                );
+                cb.require_equal(
+                    "tag_index::Rotation(3) == 1",
+                    meta.query_advice(rlp_table.tag_rindex, Rotation(3)),
+                    1.expr(),
+                );
+                cb.require_equal(
+                    "value::Rotation(3) == Rlp encoding length == index::Rotation(2)",
+                    meta.query_advice(rlp_table.value_acc, Rotation(3)),
+                    meta.query_advice(index, Rotation(2)),
+                );
+
+                // checks for RlpTxTag::Rlp on the next-to-next-to-next-to-next row.
+                cb.require_equal(
+                    "tag::Rotation(4) == RlpTxTag::Rlp",
+                    meta.query_advice(rlp_table.tag, Rotation(4)),
+                    RlpTxTag::Rlp.expr(),
+                );
+                cb.require_equal(
+                    "tag_index::Rotation(4) == tag_length::Rotation(4)",
+                    meta.query_advice(rlp_table.tag_rindex, Rotation(4)),
+                    meta.query_advice(tag_length, Rotation(4)),
+                );
+                cb.require_equal(
+                    "tag_index::Rotation(4) == 1",
+                    meta.query_advice(rlp_table.tag_rindex, Rotation(4)),
+                    1.expr(),
+                );
+                cb.require_equal(
+                    "last tag is Rlp => value_acc::Rotation(4) == value_rlc::Rotation(4)",
+                    meta.query_advice(rlp_table.value_acc, Rotation(4)),
+                    meta.query_advice(all_bytes_rlc_acc, Rotation(4)),
+                );
+                cb.require_equal(
+                    "last tag is Rlp => value_rlc::Rotation(4) == value_rlc::Rotation(2)",
+                    meta.query_advice(all_bytes_rlc_acc, Rotation(4)),
+                    meta.query_advice(all_bytes_rlc_acc, Rotation(2)),
+                );
```

Comment:
We can merge these two constraints into one `rlp_table.value_acc(4) = all_bytes_rlc_acc(2)`.